### PR TITLE
Register Page

### PR DESCRIPTION
**Pull request by allie-rae**
https://user-images.githubusercontent.com/48700332/182042352-b1f7c1d0-3238-4245-b9be-0acaf41fda03.mov

### CODE REVIEW

File: src/pages/RegisterPage/RegisterPage.tsx [hunk 1]
```diff
@@ -1,8 +1,85 @@
+import { Box, Button, Link, TextField } from '@mui/material';
+import logo from '../../assets/logo/logo.svg';
+import { Link as RouterLink } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+
 const RegisterPage: React.FC = (): React.ReactElement => {
+  const [username, setUsername] = useState('');
```
Comment (zimashima) [inline src/pages/RegisterPage/RegisterPage.tsx:7]: Why are we react-hook-form not being leveraged here? or  are we planning to refactor this and learn about it during the next meeting?

File: src/utils/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { useEffect, useRef } from 'react';
+
+// Hook
+function usePrevious<T>(value: T): T {
```
Comment (zimashima) [inline src/utils/hooks.ts:4]: What will this be used for? like in login | register ?

I propose that if it's not related to the current change, this should be a separate pull request :3 so this request will only be for any changes regarding the register page :)

File: package.json [hunk 3]
```diff
@@ -17,7 +17,6 @@
     "@types/react-dom": "^18.0.5",
     "@types/react-redux": "^7.1.24",
     "axios": "^0.27.2",
-    "mui-image": "^1.0.7",
```
Comment (zimashima) [inline package.json:20]: Bye~~~
also don't forget to remove ``@types/mui-image``

File: src/utils/hooks.ts (hunk 2 above)
Comment (allie-rae) [reply src/utils/hooks.ts:4]: Since I already did it, I'd prefer to not move it to a different PR for no reason :3

File: src/pages/RegisterPage/RegisterPage.tsx (hunk 1 above)
Comment (allie-rae) [reply src/pages/RegisterPage/RegisterPage.tsx:7]: I mentioned in the discord chat that I hadn't used react-hook-form yet and when we implement it we can change this

File: package.json (hunk 3 above)
Comment (allie-rae) [reply package.json:20]: Ooh thanks! Will do!

**Review by zimashima — CHANGES_REQUESTED** [review]
Please address ma comment :D

File: src/utils/hooks.ts (hunk 2 above)
Comment (zimashima) [reply src/utils/hooks.ts:4]: @allie-rae the reason is to conform to how version control should work should there be any changes in the future, we can follow the commit messages and revert and for us to keep track ourselves for when the changes are being incorporated :) I understand it is more convenient for us to do it this way now, but please do keep this in mind for future PR

**Comment by squashgray** [comment]
awesome login page

**Review by zimashima — APPROVED** [review]
I guess...

**Review by squashgray — APPROVED** [review]
LGTM

File: src/App.tsx [hunk 4]
```diff
@@ -2,8 +2,6 @@ import AppRoutes from './routes/AppRoutes';
 import { useGetDogByIdQuery } from './redux/dog/createApi';
```
Comment (zimashima) [inline src/App.tsx:2]:
```suggestion
```

**Review by zimashima — APPROVED** [review]
